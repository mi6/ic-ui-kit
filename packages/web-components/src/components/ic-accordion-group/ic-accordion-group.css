@import "../../global/normalize.css";

.label-container {
  padding: var(--ic-space-xs);
  display: flex;
  justify-content: space-between;
  align-items: center;
}

:host(.small) .label-container {
  padding: var(--ic-space-xxs) var(--ic-space-xs);
}

:host(.large) .label-container {
  padding: var(--ic-space-sm) var(--ic-space-xs);
}

:host(.light) {
  color: var(--ic-color-white-text);
}

:host(.accordion-group.dark) ::slotted(ic-accordion) {
  color: var(--ic-color-white-text);
}
