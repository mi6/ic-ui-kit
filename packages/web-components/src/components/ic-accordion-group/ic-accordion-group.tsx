import {
  Component,
  Element,
  h,
  Host,
  Listen,
  Prop,
  State,
} from "@stencil/core";
import { IcThemeForeground, IcSizes } from "../../utils/types";

let accordionGroupIds = 0;
@Component({
  tag: "ic-accordion-group",
  styleUrl: "ic-accordion-group.css",
  shadow: {
    delegatesFocus: true,
  },
})
export class AccordionGroup {
  private accordionGroupId = `ic-accordion-group-${accordionGroupIds++}`;

  @Element() el: HTMLIcAccordionGroupElement;

  @State() accordions: HTMLIcAccordionElement[];

  @State() areAllAccordionsOpen: boolean;

  /**
   * The accessible button label to provide more context to the 'See all/Hide all' button for screen reader users.
   */
  @Prop() accessibleButtonLabel: string = "accordions";

  /**
   * The appearance of the accordion group, e.g dark, or light.
   */
  @Prop() appearance: IcThemeForeground = "default";

  /**
   * If `true`, the accordion will load in an expanded state.
   */
  @Prop({ mutable: true }) expanded: boolean = false;

  /**
   * The header for the accordion group.
   */
  @Prop() groupTitle: string = "";

  /**
   * If `true`, only one accordion will open at a time.
   */
  @Prop() singleExpansion: boolean = false;

  /**
   * The size of the accordion.
   */
  @Prop() size?: IcSizes = "default";

  componentDidLoad(): void {
    const accordionDirectChildren = (this.el as HTMLElement).children;
    this.accordions = Array.from(accordionDirectChildren).filter(
      (child) => child.tagName === "IC-ACCORDION"
    ) as HTMLIcAccordionElement[];
    this.linkAccordions();
    this.accordions.forEach((acc) => {
      acc.appearance = this.appearance;
    });
    this.accordions.forEach((acc) => {
      acc.size = this.size;
    });
    if (this.expanded) {
      this.accordions.forEach((acc) => {
        acc.expanded = true;
      });
      this.setExpandedToAreAllAccordionsOpen();
    } else {
      this.setExpandedToAreAllAccordionsOpen();
      this.expanded = this.areAllAccordionsOpen;
    }
  }

  @Listen("accordionClicked")
  handleAccordionClicked(event: CustomEvent): void {
    if (!this.singleExpansion) {
      // 'See all' should be visible until all accordions are open, then 'Hide all' should be visible
      this.setExpandedToAreAllAccordionsOpen();
    } else {
      this.accordions.forEach((acc) => {
        if (acc.expanded && event.detail.id !== acc.id) {
          acc.expanded = false;
        }
      });
    }
  }

  private handleExpanded = () => {
    if (this.areAllAccordionsOpen) {
      this.expanded = false;
      this.accordions.forEach((acc) => {
        acc.expanded = this.expanded;
      });
    } else {
      this.expanded = true;
      this.accordions.forEach((acc) => {
        acc.expanded = this.expanded;
      });
    }
    this.setExpandedToAreAllAccordionsOpen();
  };

  private linkAccordions = () => {
    this.accordions.forEach((accordion) => {
      accordion.setAttribute("context-id", this.accordionGroupId);
    });
  };

  private setExpandedToAreAllAccordionsOpen = () => {
    this.areAllAccordionsOpen = this.accordions.every(
      (accordion) => !!accordion.expanded
    );
  };

  private accordionOpenBtnText = () => {
    return !this.areAllAccordionsOpen ? "See all" : "Hide all";
  };

  render() {
    const {
      appearance,
      size,
      groupTitle,
      singleExpansion,
      accessibleButtonLabel,
    } = this;
    return (
      <Host
        context-id={this.accordionGroupId}
        class={{
          [`${appearance}`]: true,
          [`${size}`]: true,
          ["accordion-group"]: true,
        }}
      >
        <div class="group-title-container">
          <ic-typography variant="h4">
            <h3>{groupTitle}</h3>
          </ic-typography>
          {!singleExpansion && (
            <ic-button
              appearance={appearance === "light" ? "light" : "default"}
              onClick={this.handleExpanded}
              variant="tertiary"
              aria-label={`${this.accordionOpenBtnText()} ${accessibleButtonLabel}`}
            >
              {this.accordionOpenBtnText()}
            </ic-button>
          )}
        </div>
        <slot></slot>
      </Host>
    );
  }
}
