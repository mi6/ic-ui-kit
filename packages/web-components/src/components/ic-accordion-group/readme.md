# ic-accordion-title



<!-- Auto Generated Below -->


## Properties

| Property                | Attribute                 | Description                                                                                                   | Type                              | Default        |
| ----------------------- | ------------------------- | ------------------------------------------------------------------------------------------------------------- | --------------------------------- | -------------- |
| `accessibleButtonLabel` | `accessible-button-label` | The accessible button label to provide more context to the 'See all/Hide all' button for screen reader users. | `string`                          | `"accordions"` |
| `appearance`            | `appearance`              | The appearance of the accordion group, e.g dark, or light.                                                    | `"dark" \| "default" \| "light"`  | `"default"`    |
| `expanded`              | `expanded`                | If `true`, the accordion will load in an expanded state.                                                      | `boolean`                         | `false`        |
| `groupTitle`            | `group-title`             | The header for the accordion group.                                                                           | `string`                          | `""`           |
| `singleExpansion`       | `single-expansion`        | If `true`, only one accordion will open at a time.                                                            | `boolean`                         | `false`        |
| `size`                  | `size`                    | The size of the accordion.                                                                                    | `"default" \| "large" \| "small"` | `"default"`    |


## Dependencies

### Depends on

- [ic-typography](../ic-typography)
- [ic-button](../ic-button)

### Graph
```mermaid
graph TD;
  ic-accordion-group --> ic-typography
  ic-accordion-group --> ic-button
  ic-button --> ic-loading-indicator
  ic-button --> ic-tooltip
  ic-loading-indicator --> ic-typography
  ic-tooltip --> ic-typography
  style ic-accordion-group fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


