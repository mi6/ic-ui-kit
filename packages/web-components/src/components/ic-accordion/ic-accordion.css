@import "../../global/normalize.css";

:host {
  display: block;
  border-bottom: var(--ic-border-default);
}

:host ic-typography,
:host .expand-chevron {
  color: var(--ic-color-primary-text);
}

:host(.ic-accordion-light) ic-typography,
:host(.ic-accordion-light) .expanded-content,
:host(.ic-accordion-light) .icon-container,
:host(.ic-accordion-light) .expand-chevron {
  color: var(--ic-color-white-text);
}

:host(.ic-accordion-disabled) ic-typography,
:host(.ic-accordion-disabled) .icon-container,
:host(.ic-accordion-disabled) .expand-chevron {
  color: var(--ic-architectural-500);
}

.section-button.small {
  padding: var(--ic-space-xxs) var(--ic-space-xs);
}

.section-button.large {
  padding: var(--ic-space-sm) var(--ic-space-xs);
}

:host(:first-of-type) {
  border-top: var(--ic-border-default);
}

:focus {
  outline: none;
}

.section-button {
  background-color: transparent;
  display: flex;
  align-items: center;
  width: 100%;
  padding: var(--ic-space-xs);
  font-weight: var(--ic-font-weight-bold);
  border: none;
}

.section-header {
  text-align: left;
  flex: 1 0;
}

button:hover:enabled {
  background-color: var(--ic-action-dark-bg-hover);
  cursor: pointer;
}

button:active {
  background-color: var(--ic-action-dark-bg-active);
}

button:focus {
  box-shadow: var(--ic-border-focus);
  border-radius: var(--ic-border-radius);
  transition: var(--ic-transition-duration-fast);
}

button:disabled {
  pointer-events: none;
}

.icon-container {
  margin: 0 var(--ic-space-xs) 0 0;
  display: flex;
  align-items: center;
  width: var(--ic-space-lg);
  height: var(--ic-space-lg);
}

::slotted(svg) {
  width: var(--ic-space-md);
  height: var(--ic-space-md);
}

.expand-chevron {
  width: var(--ic-space-lg);
  height: var(--ic-space-lg);
  margin-left: calc(var(--ic-space-xl) + var(--ic-space-xs));
  transform: rotate(90deg);
  justify-self: end;
}

.content-expanded-chevron {
  transform: rotate(-90deg);
}

.expanded-content {
  height: 0;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  visibility: var(--ic-expanded-content-visibility, hidden);
}

.expanded-content-inner {
  padding: var(--ic-space-xs);
}

.expanded-content-opened {
  overflow: visible;
}
