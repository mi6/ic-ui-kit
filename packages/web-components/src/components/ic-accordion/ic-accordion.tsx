import {
  Component,
  h,
  Prop,
  Element,
  Event,
  EventEmitter,
  Host,
  Watch,
  Method,
} from "@stencil/core";
import { isSlotUsed } from "../../utils/helpers";
import chevronIcon from "../../assets/chevron-icon.svg";
import { IcThemeForeground, IcSizes } from "../../utils/types";

let accordionIds = 0;

/**
 * @slot heading - Content is placed as the accordion heading.
 * @slot icon - Content is placed to the left of the heading.
 */

@Component({
  tag: "ic-accordion",
  styleUrl: "ic-accordion.css",
  shadow: true,
})
export class Accordion {
  private accordionId = `ic-accordion-${accordionIds++}`;
  private expandedContentEl: HTMLDivElement;
  private accordionBtnHeading: HTMLButtonElement;
  private CONTENT_VISIBILITY_PROPERTY = "--ic-expanded-content-visiblity";

  @Element() el: HTMLIcAccordionElement;
  /**
   * @internal Determines whether the light or dark variant of the accordion should be displayed.
   */
  @Prop() appearance?: IcThemeForeground = "default";

  /**
   * If `true`, the accordion will be disabled.
   */
  @Prop() disabled?: boolean = false;

  /**
   * If `true`, the accordion appears expanded.
   */
  @Prop({ mutable: true }) expanded: boolean = false;

  /**
   * The section header outlining section content.
   */
  @Prop() heading?: string = "";

  /**
   * The main body message of the accordion.
   */
  @Prop() message?: string = "";

  /**
   * The size of the accordion.
   */
  @Prop() size?: IcSizes = "default";

  /**
   * Emitted when accordion is clicked.
   */
  @Event() accordionClicked: EventEmitter<{ id: string }>;

  // Every time expanded is set via toggleExpanded or accordionGroup, animate to open or close
  @Watch("expanded")
  handleExpandedWatch(): void {
    this.animateExpandedContent();
  }

  /**
   * Sets focus on accordion heading.
   */
  @Method()
  async setFocus(): Promise<void> {
    if (this.accordionBtnHeading) {
      this.accordionBtnHeading.focus();
    }
  }

  disconnectedCallback(): void {
    if (this.expandedContentEl) {
      this.expandedContentEl.removeEventListener(
        "transitionend",
        (e) => this.setExpandedContentStyle(e, this.expandedContentEl),
        true
      );
      this.expandedContentEl.removeEventListener(
        "transitionend",
        (e) => this.hideExpandedContent(e, this.expandedContentEl),
        true
      );
    }
  }

  componentDidLoad(): void {
    // So accordion opens by default if expanded set to true
    if (this.expanded) {
      this.expandedContentEl.style.height = "auto";
      this.expandedContentEl.style.setProperty(
        this.CONTENT_VISIBILITY_PROPERTY,
        "visible"
      );
    }
  }

  private toggleExpanded = (): void => {
    this.expanded = !this.expanded;
    this.accordionClicked.emit({ id: this.accordionId });
  };

  // Set accordion animation
  private setAccordionAnimation = (
    el: HTMLElement,
    duration: string,
    property: string,
    delay: string
  ) => {
    el.style.transitionDuration = `${duration}ms`;
    el.style.transitionProperty = property;
    el.style.transitionDelay = delay;
  };

  private setExpandedContentStyle = (
    ev: TransitionEvent,
    expandedContent: HTMLDivElement
  ) => {
    if (ev.propertyName === "height" && expandedContent.clientHeight > 0) {
      expandedContent.classList.add("expanded-content-opened");
      expandedContent.style.height = "auto";
    }
  };

  private hideExpandedContent = (
    ev: TransitionEvent,
    expandedContent: HTMLDivElement
  ) => {
    if (ev.propertyName === "height" && expandedContent.clientHeight === 0) {
      expandedContent.style.setProperty(
        this.CONTENT_VISIBILITY_PROPERTY,
        "hidden"
      );
    }
  };

  private animateExpandedContent = () => {
    const elementHeight = this.expandedContentEl.scrollHeight;
    if (elementHeight > 0 && this.expanded) {
      this.expandedContentEl.style.setProperty(
        this.CONTENT_VISIBILITY_PROPERTY,
        "visible"
      );
      this.expandedContentEl.style.height = `${elementHeight}px`;
      this.setAccordionAnimation(
        this.expandedContentEl,
        "300",
        "height",
        "ease-out"
      );

      this.expandedContentEl.addEventListener(
        "transitionend",
        (e: TransitionEvent) => {
          this.setExpandedContentStyle(e, this.expandedContentEl);
        }
      );
    } else if (!this.expanded) {
      this.expandedContentEl.style.height = `${this.expandedContentEl.scrollHeight}px`;
      if (this.expandedContentEl.scrollHeight > 0 && !this.expanded) {
        this.expandedContentEl.style.height = "0";
        this.setAccordionAnimation(
          this.expandedContentEl,
          "300",
          "height",
          "ease-in"
        );
        this.expandedContentEl.classList.remove("expanded-content-opened");
      }
      this.expandedContentEl.addEventListener("transitionend", (e) => {
        this.hideExpandedContent(e, this.expandedContentEl);
      });
    }
  };

  render() {
    const { appearance, size, disabled, expanded } = this;
    return (
      <Host
        id={this.accordionId}
        class={{
          [`${appearance}`]: true,
          ["disabled"]: disabled,
        }}
        aria-disabled={disabled ? "true" : "false"}
      >
        <button
          ref={(el) => (this.accordionBtnHeading = el)}
          id={`${this.accordionId}-button`}
          disabled={disabled}
          tabindex={disabled ? -1 : 0}
          class={{
            [`${size}`]: true,
            ["section-button"]: true,
            ["section-button-open"]: expanded && !disabled,
          }}
          aria-expanded={`${expanded}`}
          aria-controls="expanded-content-area"
          onClick={this.toggleExpanded}
        >
          {isSlotUsed(this.el, "icon") && (
            <div class="icon-container">
              <slot name="icon" />
            </div>
          )}
          <ic-typography variant="subtitle-large" class="section-header">
            {isSlotUsed(this.el, "heading") ? (
              <slot name="heading" />
            ) : (
              this.heading
            )}
          </ic-typography>
          <span
            class={{
              ["expand-chevron"]: true,
              ["content-expanded-chevron"]: expanded && !disabled,
            }}
            aria-hidden="true"
            innerHTML={chevronIcon}
          />
        </button>
        <div
          class={{
            ["expanded-content"]: true,
          }}
          aria-labelledby={`${this.accordionId}-button`}
          role="region"
          aria-hidden={`${!expanded}`}
          id="expanded-content-area"
          ref={(el) => (this.expandedContentEl = el)}
        >
          <div class="expanded-content-inner">
            <slot />
          </div>
        </div>
      </Host>
    );
  }
}
