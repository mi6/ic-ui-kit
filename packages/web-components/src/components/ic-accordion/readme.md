# ic-accordion



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description                                   | Type                              | Default     |
| ---------- | ---------- | --------------------------------------------- | --------------------------------- | ----------- |
| `disabled` | `disabled` | If `true`, the accordion will be disabled.    | `boolean`                         | `false`     |
| `expanded` | `expanded` | If `true`, the accordion appears expanded.    | `boolean`                         | `false`     |
| `heading`  | `heading`  | The section header outlining section content. | `string`                          | `""`        |
| `message`  | `message`  | The main body message of the accordion.       | `string`                          | `""`        |
| `size`     | `size`     | The size of the accordion.                    | `"default" \| "large" \| "small"` | `"default"` |


## Events

| Event              | Description                        | Type                           |
| ------------------ | ---------------------------------- | ------------------------------ |
| `accordionClicked` | Emitted when accordion is clicked. | `CustomEvent<{ id: string; }>` |


## Methods

### `setFocus() => Promise<void>`

Sets focus on accordion heading.

#### Returns

Type: `Promise<void>`




## Slots

| Slot        | Description                                   |
| ----------- | --------------------------------------------- |
| `"heading"` | Content is placed as the accordion heading.   |
| `"icon"`    | Content is placed to the left of the heading. |


## Dependencies

### Depends on

- [ic-typography](../ic-typography)

### Graph
```mermaid
graph TD;
  ic-accordion --> ic-typography
  style ic-accordion fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


