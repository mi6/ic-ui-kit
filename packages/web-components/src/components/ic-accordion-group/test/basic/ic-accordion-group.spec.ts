import { newSpecPage } from "@stencil/core/testing";
import { AccordionGroup } from "../../ic-accordion-group";
import { Accordion } from "../../../ic-accordion/ic-accordion";

const event = { detail: { id: "ic-accordion-0" } };

beforeAll(() => {
  jest.spyOn(console, "warn").mockImplementation(jest.fn());
});

describe("ic-accordion-group snapshots", () => {
  it("should match default snapshot", async () => {
    const page = await newSpecPage({
      components: [AccordionGroup],
      html: `<ic-accordion-group group-title="Test heading"></ic-accordion-group>`,
    });
    expect(page.root).toMatchSnapshot("renders as default");
  });

  it("should match light snapshot", async () => {
    const page = await newSpecPage({
      components: [AccordionGroup],
      html: `<ic-accordion-group appearance="light" group-title="Test heading"></ic-accordion-group>`,
    });
    expect(page.root).toMatchSnapshot("renders as light");
  });
});

describe("ic-accordion-group functions unit test", () => {
  it("checks setExpandedToAreAllAccordionsOpen when single expansion is false", async () => {
    const page = await newSpecPage({
      components: [AccordionGroup, Accordion],
      html: `
    <ic-accordion-group group-title="Test heading">
      <ic-accordion>
        <ic-typography variant="body" >
          This is an example of the main body text.
        </ic-typography>
      </ic-accordion>
      <ic-accordion expanded>
      <ic-typography variant="body" >
        This is an example of the main body text.
      </ic-typography>
    </ic-accordion>
    </ic-accordion-group>`,
    });
    const spySetExpandedToAreAllAccordionsOpen = jest.spyOn(
      page.rootInstance,
      "setExpandedToAreAllAccordionsOpen"
    );
    expect(page.root.singleExpansion).toBe(false);
    await page.rootInstance.handleAccordionClicked(event);
    expect(spySetExpandedToAreAllAccordionsOpen).toBeCalled();
  });

  it("checks handleExpanded", async () => {
    const page = await newSpecPage({
      components: [AccordionGroup],
      html: `<ic-accordion-group group-title="Test heading" expanded></ic-accordion-group>`,
    });
    expect(page.root.expanded).toBe(true);
    await page.rootInstance.handleExpanded();
    await page.waitForChanges;
    expect(page.root.expanded).toBe(false);
  });
});

describe("ic-accordion-group component", () => {
  it("it should test the handleExpanded function", async () => {
    const page = await newSpecPage({
      components: [AccordionGroup, Accordion],
      html: `
      <ic-accordion-group expanded="true" group-title="Test heading">
        <ic-accordion heading="Accordion 1">
          <ic-typography variant="body">
            This is an example of the main body text.
          </ic-typography>
        </ic-accordion>
      </ic-accordion-group>`,
    });
    expect(page.root.expanded).toBe(true);
    await page.rootInstance.handleExpanded();
    await page.waitForChanges();
    expect(page.root.expanded).toBe(false);
  });

  it("it should test singleExpansion function", async () => {
    const page = await newSpecPage({
      components: [AccordionGroup, Accordion],
      html: `
      <ic-accordion-group group-title="Test heading" single-expansion="true">
        <ic-accordion heading="Accordion 1">
          <ic-typography variant="body">
            This is an example of the main body text.
          </ic-typography>
        </ic-accordion>
        <ic-accordion heading="Accordion 2" expanded>
          <ic-typography variant="body">
            This is an example of the main body text.
          </ic-typography>
        </ic-accordion>
      </ic-accordion-group>`,
    });

    const accordions = page.root.querySelectorAll("ic-accordion");
    const accordion1 = accordions[0];
    const accordion2 = accordions[1];
    const accordionButton = accordion1.shadowRoot.querySelector(
      ".section-button"
    ) as HTMLButtonElement;
    await page.waitForChanges();
    expect(accordion1.expanded).toBe(false);
    expect(accordion2.expanded).toBe(true);
    accordionButton.click();
    await page.waitForChanges();
    expect(accordion1.expanded).toBe(true);
    expect(accordion2.expanded).toBe(false);
  });

  it("checks the areAllAccordionsOpen function", async () => {
    const page = await newSpecPage({
      components: [AccordionGroup, Accordion],
      html: `
      <ic-accordion-group group-title="Test heading">
        <ic-accordion heading="Accordion 1">
          <ic-typography variant="body">
            This is an example of the main body text.
          </ic-typography>
        </ic-accordion>
      </ic-accordion-group>`,
    });
    // uses handleExpanded to open accordion, making areAllAccordionsOpen = true
    await page.rootInstance.handleExpanded();
    await page.waitForChanges;
    expect(page.root.expanded).toBe(true);
    expect(page.rootInstance.areAllAccordionsOpen).toBe(true);

    // uses handleExpanded to close accordion, making areAllAccordionsOpen = false
    await page.rootInstance.handleExpanded();
    await page.waitForChanges;
    expect(page.root.expanded).toBe(false);
    expect(page.rootInstance.areAllAccordionsOpen).toBe(false);
  });
});
