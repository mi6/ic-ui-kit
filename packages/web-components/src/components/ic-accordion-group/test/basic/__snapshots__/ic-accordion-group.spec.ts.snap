// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-accordion-group snapshots should match default snapshot: renders as default 1`] = `
<ic-accordion-group class="accordion-group default" context-id="ic-accordion-group-0" group-title="Test heading">
  <mock:shadow-root>
    <div class="group-title-container">
      <ic-typography variant="h4">
        <h3>
          Test heading
        </h3>
      </ic-typography>
      <ic-button appearance="default" aria-label="Hide all accordions" variant="tertiary">
        Hide all
      </ic-button>
    </div>
    <slot></slot>
  </mock:shadow-root>
</ic-accordion-group>
`;

exports[`ic-accordion-group snapshots should match light snapshot: renders as light 1`] = `
<ic-accordion-group appearance="light" class="accordion-group default light" context-id="ic-accordion-group-1" group-title="Test heading">
  <mock:shadow-root>
    <div class="group-title-container">
      <ic-typography variant="h4">
        <h3>
          Test heading
        </h3>
      </ic-typography>
      <ic-button appearance="light" aria-label="Hide all accordions" variant="tertiary">
        Hide all
      </ic-button>
    </div>
    <slot></slot>
  </mock:shadow-root>
</ic-accordion-group>
`;
